### PR TITLE
2903 currency selection for outbound shipment

### LINKED ISSUE #2903: Currency selection for Outbound Shipment
## Is your feature request related to a problem? Please describe 👀
Users should be able to select a different currency in outbound shipments.

## Describe the solution you'd like 🎁
Add a currency select component to outbound shipment and when users select a different currency, the prices in the invoice should change accordingly.

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #2903

# 👩🏻‍💻 What does this PR do? 
Allow users to select foreign currencies in outbound shipments... not sure about having the scroll on line edit view though...
Screenshot 2024-02-08 at 13 29 53
Screenshot 2024-02-08 at 13 30 04
Screenshot 2024-02-08 at 13 30 16

# 🧪 How has/should this change been tested? 
- [ ] Must have store preference `Store: Able to issue in foreign currency` on
- [ ] Go to outbound shipments
- [ ] Add some lines
- [ ] Change currency (See side panel for prices)
- [ ] Click on a line and see the foreign currency price conversions next to sell price

## 📃 Documentation
Documentation for outbound shipment foreign currency

### CODE REVIEW

File: client/packages/invoices/src/OutboundShipment/DetailView/SidePanel/PricingSection.tsx [hunk 1]
```diff
@@ -141,6 +149,52 @@ const ItemPrices = ({ pricing, isDisabled }: PricingGroupProps) => {
...
+          {t('heading.foreign-currency')}
+        </PanelLabel>
+        <PanelField>
+          <CurrencyModal
+            onChange={onChange}
+            isDisabled={
+              otherPartyIsInternal || !store?.preferences.issueInForeignCurrency
+            }
+            currency={currency as CurrencyRowFragment}
+            currencyRate={currencyRate ?? 1}
```
Comment (Chris-Petty) [inline client/packages/invoices/src/OutboundShipment/DetailView/SidePanel/PricingSection.tsx:176]: Might be late in the game but... can we make currency guaranteed? If the answer is yes I'd do it in another issue. All invoices SHOULD at least have the default currency as default currency, so shouldn't be optional really.

File: server/service/src/invoice/outbound_shipment/update/validate.rs [hunk 2]
```diff
@@ -28,7 +28,7 @@ pub fn validate(
     if !check_invoice_type(&invoice, InvoiceRowType::OutboundShipment) {
         return Err(NotAnOutboundShipment);
     }
-    if patch.currency_id.is_some() && !check_can_issue_in_foreign_currency(connection, store_id)? {
+    if patch.currency_id.is_some() && !check_can_issue_in_foreign_currency(connection, store_id)? && other_party.store_row.is_some() {
```
Comment (Chris-Petty) [inline server/service/src/invoice/outbound_shipment/update/validate.rs:31]: There is no `other_party` variable in scope. Strange that rust analyser isn't picking it up. Compiling does at least.

**Review by Chris-Petty — CHANGES_REQUESTED** [review]
Looks good, was gonna give a quick whirl but it no compile 😱. Probs because upsteam changes merged in removed stuff.

File: client/packages/invoices/src/OutboundShipment/DetailView/SidePanel/PricingSection.tsx (hunk 1 above)
Comment (roxy-dao) [reply client/packages/invoices/src/OutboundShipment/DetailView/SidePanel/PricingSection.tsx:176]: #3134

**Pull request merged**